### PR TITLE
Support legacy KafkaChannel secret format

### LINKED ISSUE #2213: Migration to new KafkaChannel does take into account authentication defaults from old implementation
**Describe the bug**
The previous implementation of the channel did not require `protocol` and `sasl.mechanism` to be present in the Secret that was used for authentication.
The `protocol` settings is not required otherwise the Broker will fail with this error:
```
reason: 'error getting cluster admin config: protocol required (key: protocol)
      supported protocols: [PLAINTEXT SASL_PLAINTEXT SSL SASL_SSL]'
```
The sasl.mechanism now defaults to PLAIN as implemented in https://github.com/knative-sandbox/eventing-kafka-broker/issues/2117 but if the previous implementation was using some other mechanism, the new default (PLAIN) will not work with the existing Kafka cluster. This effectively breaks the existing deployments and the KafkaChannel stops working during the upgrade.

 **Expected behavior**
The protocol and sasl.mechanism is properly migrated when migrating from old to new channel implementation.

 **To Reproduce**
Assuming a Strimzi-installed Kafka in kafka namespace with:
```
       - name: sasl
         port: 9094
         type: internal
         tls: true
         authentication:
           type: scram-sha-512
```
and a KafkaUser my-sasl-user user.

* In a knative-eventing namespace, create the secret that will be used by KnativeKafka.
```
STRIMZI_CRT=$(oc -n kafka get secret my-cluster-cluster-ca-cert --template='{{index .data "ca.crt"}}' | base64 --decode )
SASL_PASSWD=$(oc -n kafka get secret my-sasl-user --template='{{index .data "password"}}' | base64 --decode )
oc create secret --namespace knative-eventing generic my-sasl-secret \
      --from-literal=ca.crt="$STRIMZI_CRT" \
      --from-literal=password="$SASL_PASSWD" \
      --from-literal=saslType="SCRAM-SHA-512" \
      --from-literal=user="my-sasl-user"
```
The secret above works fine with the previous implementation of the Channel.

* Now install the previous version of KnativeEventing and KnativeKafka with the following configs:
kafka-channel-config:
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: kafka-channel-config
  namespace: knative-eventing
data:
  auth.secret.ref.name: my-sasl-secret
  auth.secret.ref.namespace: knative-eventing
  bootstrap.servers: 'my-cluster-kafka-bootstrap.kafka:9094'
```
config-br-default-channel:
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: config-br-default-channel
  namespace: knative-eventing
data:
  channel-template-spec: |-
    apiVersion: messaging.knative.dev/v1beta1
    kind: KafkaChannel
    spec:
      numPartitions: 6
      replicationFactor: 3
      delivery:
        retry: 12
        backoffPolicy: exponential
        backoffDelay: PT1S
```
config-br-defaults:
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: config-br-defaults
  namespace: knative-eventing
data:
  default-br-config: |
    clusterDefault:
      apiVersion: v1
      brokerClass: MTChannelBasedBroker
      delivery:
        backoffDelay: PT0.2S
        backoffPolicy: exponential
        retry: 10
      kind: ConfigMap
      name: config-br-default-channel
      namespace: knative-eventing
```
* Create a channel like this:
```
apiVersion: eventing.knative.dev/v1
kind: Broker
metadata:
  name: my-broker
  namespace: my-namespace
```
* Upgrade Eventing to the latest version which uses the new implementation.
* The channel ends up with error like this:
```
apiVersion: messaging.knative.dev/v1beta1
kind: KafkaChannel
metadata:
  annotations:
    eventing.knative.dev/scope: cluster
  creationTimestamp: "2022-05-23T08:47:23Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2022-05-23T08:57:39Z"
  finalizers:
  - kafkachannels.messaging.knative.dev
  generation: 2
  labels:
    eventing.knative.dev/broker: my-broker
    eventing.knative.dev/brokerEverything: "true"
  name: my-broker-kne-trigger
  namespace: serverless-tests
  ownerReferences:
  - apiVersion: eventing.knative.dev/v1
    blockOwnerDeletion: true
    controller: true
    kind: Broker
    name: my-broker
    uid: ef34b9f7-acc7-44bd-affb-31e42d5fc2c5
  resourceVersion: "125460"
  uid: 966c5534-2fb1-4e6b-9279-8f0fc402dfad
spec:
  delivery:
    backoffDelay: PT1S
    backoffPolicy: exponential
    retry: 12
  numPartitions: 6
  replicationFactor: 3
status:
  conditions:
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    status: Unknown
    type: Addressable
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    status: Unknown
    type: ConfigMapUpdated
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    status: "True"
    type: ConfigParsed
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    status: "True"
    type: DataPlaneAvailable
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    status: Unknown
    type: ProbeSucceeded
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    message: 'error getting cluster admin sarama config: protocol required (key: protocol)
      supported protocols: [PLAINTEXT SASL_PLAINTEXT SSL SASL_SSL]'
    reason: 'Failed to create topic: knative-messaging-kafka.serverless-tests.my-broker-kne-trigger'
    status: "False"
    type: Ready
  - lastTransitionTime: "2022-05-23T08:47:23Z"
    message: 'error getting cluster admin sarama config: protocol required (key: protocol)
      supported protocols: [PLAINTEXT SASL_PLAINTEXT SSL SASL_SSL]'
    reason: 'Failed to create topic: knative-messaging-kafka.serverless-tests.my-broker-kne-trigger'
    status: "False"
    type: TopicReady
  observedGeneration: 1
```

Note: If the secret was created like below the new implementation works as expected after upgrade:
```
oc create secret --namespace knative-eventing generic my-sasl-secret \
      --from-literal=ca.crt="$STRIMZI_CRT" \
      --from-literal=password="$SASL_PASSWD" \
      --from-literal=saslType="SCRAM-SHA-512" \
      --from-literal=user="my-sasl-user" \
      --from-literal=protocol="SASL_SSL" \  #This is added.
      --from-literal=sasl.mechanism="SCRAM-SHA-512" # This is added.
```

 **Knative release version**
1.2.3

### PR DESCRIPTION
**Pull request by pierDipi**
Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

Fixes #2213

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/channel/channel_test.go [hunk 1]
```diff
@@ -1062,7 +1062,7 @@ func TestReconcileKind(t *testing.T) {
 			},
 		},
 		{
-			Name: "Reconciled normal - with single fresh subscriber - with auth - PlainText",
+			Name: "Reconciled normal - with single fresh subscriber - with auth - SSL",
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/channel/channel_test.go:1065]: This test was using a SSL secret but the name contained PlainText, so renamed :)

**Comment by aliok** [comment]
Let's first merge https://github.com/openshift-knative/eventing-kafka-broker/pull/283

**Review by aliok — APPROVED** [review]
/lgtm
/approve

Verified with the instructions at https://github.com/knative-sandbox/eventing-kafka-broker/issues/2213 locally

**Comment by pierDipi** [comment]
/cc @aliok 
(I had to force push since CLA check was stuck)

**Review by aliok — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**